src/NUnitTestExtractor/NUnitTestExtractorApp.cs

```diff
@@ -16,383 +15,264 @@ public static class NUnitTestExtractorApp
-        /// <param name="type">Used to get the namespace of the test</param>
-        /// <param name="methodInfo">Used to get the test info such as declaring type and name</param>
-        /// <param name="testCase">Used to get arguments of a testcase</param>
-        /// <param name="data">The string which is modified and then returned</param>
-        /// <returns>Returns properly formatted test string</returns>
-        private static string FormattedTest(Level level, Type type, MethodInfo methodInfo, TestCaseAttribute testCase, string data)
+        /// <param name="assemblyFullPath"></param>
+        /// <returns></returns>
+        public static void GenerateInfoFromAssembly(string assemblyFullPath)
         {
-            switch (level)
-            {
-                case Level.Namespace:
-                    data = type.Namespace;
-                    break;
+            HashSet<string> rv = new HashSet<string>();
 
-                case Level.Class:
-                    data = methodInfo.DeclaringType.ToString();
-                    break;
-
-                case Level.Function:
-                    data = methodInfo.DeclaringType + "." + methodInfo.Name;
-                    break;
-
-                case Level.TestCase:
-                    if (testCase != null)
-                    {
-                        data = FormattedTestCase(methodInfo.DeclaringType.ToString(), methodInfo.Name, testCase.Arguments, data);
-                    }
+            foreach (var item in GetValidTestSuiteTypesFromAssembly(assemblyFullPath))
+            {
+                Level lv = ParseLevel(_options.Level);
 
-                    break;
+                switch (lv)
+                {
+                    case Level.Namespace:
+                        rv.Add(item.Namespace);
+                        break;
+                    case Level.Class:
+                        rv.Add(item.FullName);
+                        break;
+                    case Level.Function:
+                        rv.UnionWith(GetValidTestsFromTestSuite(item));
+                        break;
+                    case Level.TestCase:
+                        break;
+                    default:
+                        throw new ArgumentException(StringUtils.FormatInvariant("Invalid level option: {0}", lv.ToString()));
+                }
             }
 
-            return data;
+            OutputInfo(rv);
         }
 
         /// <summary>
-        /// Add testCase categories to categoryNames 
+        /// Load assembly by passing its path
         /// </summary>
-        /// <param name="categoryNames">list to add to</param>
-        /// <param name="testCase">testCase used to get arguments from</param>
-        private static void AddTestCaseCategoriesToList(List<string> categoryNames, TestCaseAttribute testCase)
+        /// <param name="assemblyFullPath"></param>
+        /// <returns></returns>
```

Comment:
[Should] Remove the returns tag & fill in the param tag.

Reply:
This function is no longer exist